### DISCUSSION TITLE
navigation config via wpf-ui

### DISCUSSION THREAD

**Discussion by cbptamtom** [Nice3point/RevitTemplates#51] [Ideas]
Hi Roman. Thanks for all your help. It would be great if you could create a new simple project that demonstrates how to configure navigation for multiple pages using WPF UI library.

**Comment by Nice3point** [comment]
Hi) RevitLookup is not very informative for you? It uses navigation. And the examples themselves from the wpf-ui repository

https://github.com/lepoco/wpfui/tree/main/src

You just need to implement `INavigableView<PageViewModelType>` for your page. And set `TargetPageType` for `NavigationViewItem`

```xaml
<ui:NavigationView.MenuItems>
    <ui:NavigationViewItem
          Content="Dashboard"
          TargetPageType="{x:Type pages:DashboardPage}"/>
</ui:NavigationView.MenuItems>
```

**Comment by cbptamtom** [comment]
yeah thanks Roman,
I got that and finished that so easy.

![image](https://github.com/Nice3point/RevitTemplates/assets/133786016/34415c8c-cd4e-49e5-bd32-28f4860f4e3b)
![image](https://github.com/Nice3point/RevitTemplates/assets/133786016/b6645c93-84c8-44ad-8c49-4cf883efbc45)

**Reply by Nice3point** [reply]
great, congrats)

**Comment by cbptamtom** [comment]
I can using navigation too smooth but have a problem when connect page with viewmodel of page. It's really a difficult challenge

**Reply by Nice3point** [reply]
Why? You have to create a new Viewmodel instance and set DataContext for page. Thats it

**Reply by cbptamtom** [reply]
With this case, I can navigate that. But I didn't use the button click action; it didn't perform any action at all. And if I add the ViewModel to the code-behind of the LinesView, I think it would trigger the event, but it doesn't; it doesn't navigate there at all.

![image](https://github.com/Nice3point/RevitTemplates/assets/133786016/1e716ab1-03fa-478f-9c46-8aed87c34f52)
![image](https://github.com/Nice3point/RevitTemplates/assets/133786016/0b5e154a-dbc7-4252-8214-1302bc0656f7)
![image](https://github.com/Nice3point/RevitTemplates/assets/133786016/a1d93fcc-5acd-4c61-9551-22b843e79606)

**Reply by Nice3point** [reply]
d:DataContext uses only for code highlighting in xaml. Thats is.
For runtime you need to set `DataContext` property in View.xaml.cs before `InitializeComponet`

```C#
public LinesView(LinesViewModel viewmodel)
{
   DataContext = viewmodel;
   InitializeComponets();
}
```

**Reply by cbptamtom** [reply]
I also tried that method, but it still didn't work for me. I will look into your configuration approach in RevitLookup.

**Reply by cbptamtom** [reply]
![image](https://github.com/Nice3point/RevitTemplates/assets/133786016/8058bdda-06b9-4b9c-b041-5f9c49718737)
Done that. Thank Roman so much. I got that according the RevitLookup opensource. Thanks

**Reply by Nice3point** [reply]
That's perfect) It's great that it worked out for you
